espresso/environment/9_pipeline_enhancement_test.go

```diff
@@ -0,0 +1,98 @@
+package environment_test
+
+import (
+	"context"
+	env "github.com/ethereum-optimism/optimism/espresso/environment"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-e2e/system/e2esys"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/stretchr/testify/require"
+	"io"
+	"log/slog"
+	"math/big"
+	"testing"
+)
+
+// TestPipelineEnhancement is a test that ensures the derivation pipeline does not include batches from reverted L1 transactions submitted to the inbox contract.
+// Attempt to post a batch that fails in the batch inbox contract. The revert transactions should not be included by the derivation pipeline.
+//		Arrange:
+//			Running Sequencer, Batcher in Espresso mode
+//		Act:
+//			Send a transaction not signed by the batcher (which means it will revert) to the inbox contract with a specific sequence of bytes e.g. 0x42424242424242424242
+//          Fetch N, the block number where the transaction was included (even though it is reverted)
+//		Assert:
+//			Instantiate a CalldataSource object with block N. CalldataSource is the object that reads the calldata from the inbox contract and filters it using the isValidBatchTx function.
+//			Verify that the concatenated bytes of all batch transactions does not include 0x42424242424242424242
+
+func TestPipelineEnhancement(t *testing.T) {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	launcher := new(env.EspressoDevNodeLauncherDocker)
+
+	system, espressoDevNode, err := launcher.StartDevNet(ctx, t)
+	// Signal the testnet to shut down
+	if have, want := err, error(nil); have != want {
```

Comment:
Nit: this would be more readable as 
`require.NoError(t, err, "failed to start dev environment with espresso dev node")`.

Reply:
Yes and also more consistent with the rest of the test. Change done in 1cae1af35.